assets/main.js

```diff
@@ -529,3 +529,106 @@ async function logout() {
 		buttonLoading(logoutButton, false, "Logout")
 	}
 }
+
+async function deleteVolume(block_name) {
+	const deleteButton = document.getElementById(`delete-block-button-${block_name}`);
+	const molly_csrf = document.getElementById("molly-csrf").value;
+	const formAlert = document.getElementById("form-alert");
+	try {
+		buttonLoading(deleteButton, true, "Deleting...")
+		const response = await fetch("/api/volume/delete", {
+			method: 'POST',
+			headers: {
+				"Content-Type": "application/json",
+			},
+			body: JSON.stringify(
+				{
+					"block_name": block_name,
+					"molly_csrf": molly_csrf
+				})
+		})
+		const data = await response.json();
+		if (data.status === 200) {
+			formAlert.classList.remove("hidden", "text-secondary-500");
+			formAlert.classList.add("text-primary-500");
+			formAlert.textContent = "Succesfully deleted";
+			postAlert("bg-primary-300", "Volume deleted succesfully");
+			setTimeout(() => window.location.reload(), 1000);
+			buttonLoading(deleteButton, false, "Delete")
+		} else {
+			formAlert.classList.remove("hidden", "text-primary-500");
+			formAlert.classList.add("text-secondary-500");
+			formAlert.textContent = data.data
+			buttonLoading(deleteButton, false, "Delete")
+		}
+	} catch (error) {
+		formAlert.classList.remove("hidden", "text-primary-500");
+		formAlert.classList.add("text-secondary-500");
+		formAlert.textContent = error
+		buttonLoading(deleteButton, false, "Delete")
+	}
+}
+
+async function createVolume() {
+	const createButton = document.getElementById("create-block-button");
+	const block_name = document.getElementById("block_name").value;
+	const block_size = document.getElementById("block_size").innerText;
+	const data_toggle = document.getElementById("dataToggle").checked;
+	const molly_csrf = document.getElementById("molly-csrf").value;
+	const formAlert = document.getElementById("form-alert");
+	const block_compressed = document.getElementById("block_compressed").checked;
+	const block_data = document.getElementById("block_data").files[0];
+	try {
+		if (block_name === "") {
```

Comment:
I wonder whether we should in JavaScript already have functions for valid names (e.g. when creating a block device, but also when creating a unikernel).

FWIW, from albatross vmm_core:
```OCaml
  let [@inline always] valid_label s =
    String.length s < 64 &&
    String.length s > 0 &&
    String.get s 0 <> '-' && (* leading may not be '-' *)
    for_all (function
        | 'a'..'z' | 'A'..'Z' | '0'..'9' | '-' | '.' -> true
        | _ -> false)
      s (* only LDH (letters, digits, hyphen, period)! *)
```

So, between 1 and 63 characters in length, and only a-zA-Z0-9 and - and . are allowed. The first character may not be "-". I wonder whether we should actually use js_of_ocaml to compile some of the albatross bits and pieces to JavaScript (maybe at a later point in time).

Reply:
On the server side, we already do somewhere (since we translate that string to a Vmm_core.Name.t)

Reply:
fixed in [de852fd](https://github.com/robur-coop/mollymawk/pull/90/commits/de852fdad634f47955273ceb94811d4a4e01d1b5)